app/assets/javascripts/components/home_page.jsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import * as ReactRedux from 'react-redux';
+
+// Module imports.
+import * as TimurActions from '../../javascripts/actions/timur_actions';
+
+export class HomePage extends React.Component{
+  constructor(props){
+    super(props);
+    this.state = {};
+  }
+
+  componentDidMount(){
+    this.props.fetchPermissionsSettings();
+  }
+
+  renderProjects(){
+    if(!this.props.token_permissions){
+      return <span>{'No available projects.'}</span>;
+    }
+
+    let link_props = {className: 'home-page-project-link'};
+    return this.props.token_permissions.map((permission, index)=>{
+      link_props['key'] = index;
+      link_props['href'] =  '/' + permission.project_name;
+      return(
+        <a {...link_props}>
+
+          {permission.project_name} &nbsp;
+          <span className='home-page-project-role'>({permission.role})</span>
+        </a>
+      );
+    });
+  }
+
+  render(){
+    return(
+      <div className='browser project'>
+
+        <div className='home-page-header-group'>
+
+          <div className='page-detail-group'>
+
+            {'Available Projects'}
+          </div>
+        </div>
+        <div className='browser-tab'>
+
+          {this.renderProjects()}
+        </div>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state = {}, own_props)=>{
+  if(state.timur.user === undefined) return {};
```

Comment:
let's try to make a selector for this, i'd like to avoid decomposing the state directly in these connector functions

Reply:
If the data needs to be mutated or it's nested deeply then I'll write a selector.

Reply:
The value of the selector is not only in easing the extraction, it is (especially) in abstracting away the state from the component, so the component does not interact in any way with the state structure, it just requests information from the state via a function. For example, in this case if I want to move the user model out of state.timur up to state.user, I would have to rewrite this component and every other component that touches the user state. However, if the component merely calls selectUser() or  select('USER'), we only have to update the selector. For this reason I think it makes sense to always use selectors to extract data from the state.